model/src/test/kotlin/licenses/LicenseConfigurationTest.kt

```diff
@@ -0,0 +1,144 @@
+            val cat2 = LicenseCategory("non permissive", "Strict licenses")
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), sortedSetOf("permissive"),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val lic3 = License(
+                SpdxSingleLicenseExpression.parse("GPL"), sortedSetOf("non permissive"),
+                includeInNoticeFile = false, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(
+                licenseCategories = listOf(cat1, cat2),
+                licenses = listOf(lic1, lic2, lic3)
+            )
+
+            val permissiveLicenses = licenseConfiguration.getLicensesForCategory(cat1.name)
+
+            permissiveLicenses shouldContainExactlyInAnyOrder setOf(lic1, lic2)
+        }
+
+        "throw an exception when querying the licenses for an unknown category" {
+            val cat = LicenseCategory("oneAndOnlyCategory")
+            val lic = License(
+                SpdxSingleLicenseExpression.parse("LICENSE"), sortedSetOf(cat.name),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(licenses = listOf(lic), licenseCategories = listOf(cat))
+
+            shouldThrow<IllegalStateException> {
+                licenseConfiguration.getLicensesForCategory("nonExistingCategory")
+            }
+        }
+
+        "allow querying a license by ID" {
+            val lic1 = License(
+                SpdxSingleLicenseExpression.parse("ASL-1"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = false
+            )
+            val lic2 = License(
+                SpdxSingleLicenseExpression.parse("ASL-2"), emptySortedSet(),
+                includeInNoticeFile = true, includeSourceCodeOfferInNoticeFile = true
+            )
+            val licenseConfiguration = LicenseConfiguration(licenses = listOf(lic1, lic2))
+
+            licenseConfiguration[SpdxSingleLicenseExpression.parse("ASL-2")] shouldBe lic2
+        }
+
+        "allow access to the names of all categories" {
+            val cat1 = LicenseCategory("permissive", "Permissive licenses")
+            val cat2 = LicenseCategory("non permissive", "Strict licenses")
+            val cat3 = LicenseCategory("other", "Completely different licenses")
+            val licenseConfiguration = LicenseConfiguration(licenseCategories = listOf(cat1, cat2, cat3))
+
+            licenseConfiguration.categoryNames shouldContainExactly listOf("non permissive", "other", "permissive")
```

Comment:
How about `shouldContainExactlyInAnyOrder` ?

Reply:
As the category names are a sorted set, I thought it may make sense to check the order as well.

Reply:
I fine as-is as well.